zio-http/src/main/scala/zhttp/http/Proxy.scala

```diff
@@ -0,0 +1,37 @@
+package zhttp.http
+import io.netty.handler.codec.http.EmptyHttpHeaders
+import io.netty.handler.proxy.HttpProxyHandler
+import zhttp.http.middleware.Auth.Credentials
+
+import java.net.InetSocketAddress
+
+final case class Proxy(
```

Comment:
May be we can add a bit of scala doc for the users to understand the usage of `Proxy` a little better.